### PR TITLE
[PBW-3460] makes postroll ad play before upNext video if upNext is started by the timer.

### PR DESCRIPTION
**Pull request by ksunika**

### CODE REVIEW

File: js/components/countDownClock.js [hunk 1]
```diff
@@ -156,7 +156,8 @@ var CountDownClock = React.createClass({
       "clickedVideo" : this.props.upNextInfo.upNextData,
       "custom" : {"source": CONSTANTS.SCREEN.UP_NEXT_SCREEN}
     };
-    this.props.controller.sendDiscoveryClickEvent(eventData);
+    var isAutoUpNext = true;
+    this.props.controller.sendDiscoveryClickEvent(eventData, isAutoUpNext);
   },
```
Comment (chibzu) [inline js/components/countDownClock.js]: Does startDiscoveryVideo need this flag also?

File: js/controller.js [hunk 2]
```diff
@@ -620,10 +629,16 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
       }
     },
 
-    sendDiscoveryClickEvent: function(selectedContentData) {
+    sendDiscoveryClickEvent: function(selectedContentData, isAutoUpNext) {
```
Comment (chibzu) [inline js/controller.js]: All of the calls to this method should be updated to include "false" so we aren't relying on using undefined as false.

Comment (ksunika) [inline js/controller.js]: done!

File: js/components/countDownClock.js (hunk 1 above)
Comment (aldigun) [inline js/components/countDownClock.js]: @ksunika on Zhibzu comment.

File: js/components/countDownClock.js [hunk 3]
```diff
@@ -156,7 +156,8 @@ var CountDownClock = React.createClass({
       "clickedVideo" : this.props.upNextInfo.upNextData,
       "custom" : {"source": CONSTANTS.SCREEN.UP_NEXT_SCREEN}
     };
-    this.props.controller.sendDiscoveryClickEvent(eventData);
+    var isAutoUpNext = true;
```
Comment (aldigun) [inline js/components/countDownClock.js]: We can remove this line and set sendDiscoveryClickEvent(eventData, true)

File: js/controller.js [hunk 4]
```diff
@@ -58,7 +58,9 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
         "countDownFinished": false,
         "countDownCancelled": false,
         "timeToShow": 0,
-        "showing": false
+        "showing": false,
+        "delayedSetEmbedCodeEvent": false,
+        "selectedContentData": null
```
Comment (aldigun) [inline js/controller.js]: - Weird naming, "selectedContentData" can be confusing to developer.
- selectedContentData is only being used if delayedSetEmbedCodeEvent is set to TRUE. Do you agree to refactor these 2 variables?

Comment (aldigun) [inline js/controller.js]: After re-thinking it, I don't like the other possibility even more.
Let's stays as is for 2 variables. But at least reset selectedContentData after it is being consumed.

Comment (ksunika) [inline js/controller.js]: Ok, I will rename selectedContentData to delayedContentData and reset it after consumed.

File: js/components/countDownClock.js (hunk 1 above)
Comment (ksunika) [inline js/components/countDownClock.js]: @chibzu, the problem was that postroll did not play if upNext timer was up. 
For Discover, we first see the postroll and only then the Discover screen with the timer appear, so the issue does not occur.

File: js/controller.js (hunk 4 above)
Comment (ksunika) [inline js/controller.js]: @aldigun, done

**Comment by aldigun** [comment]
LGTM

**Pull request merged**